### PR TITLE
Add test and fix for categorical series .shift #10495.

### PR DESCRIPTION
**Pull request by cottrell**
#10495

Waiting for Travis and will review further.

### CODE REVIEW

File: pandas/core/internals.py [hunk 1]
```diff
@@ -887,7 +887,11 @@ def shift(self, periods, axis=0):
         # that, handle boolean etc also
         new_values, fill_value = com._maybe_upcast(self.values)
         # make sure array sent to np.roll is c_contiguous
-        f_ordered = new_values.flags.f_contiguous
+
+        f_ordered = False
+        if not isinstance(new_values, Categorical):
```
Comment (jreback) [inline pandas/core/internals.py]: just define ``.flags`` on ``Categorical``

**Comment by sinhrks** [comment]
Pls add release note to ``doc/source/whatsnew/v0.17.0.txt``

File: pandas/tests/test_categorical.py [hunk 2]
```diff
@@ -1175,6 +1175,17 @@ def test_comparison_with_unknown_scalars(self):
...
 
+    def test_shift(self):
+        # GH10495
+        # Series.shift should not depend on the dtype being categorical or not
+        values = [['a', 'b', 'c']]
+        shifts = [-1, 0, 1]
+        for v in values:
+            for shift in shifts:
+                a = pd.Series(v).shift(shift)
+                b = pd.Series(v, dtype='category').shift(shift)
```
Comment (jreback) [inline pandas/tests/test_categorical.py]: this needs to assert that the result series is equal 
so construct the expected result as series

**Comment by jreback** [comment]
#10497 fixes this i believe. But pls check.

**Comment by jreback** [comment]
#10497 is a much better way to do this

**Pull request closed without merging**